### PR TITLE
Add cookiecutter (and jinja2-time, whichcraft, binaryornot, poyo, arrow)

### PR DESCRIPTION
**Pull request by bollwyvl**
This adds [cookiecutter](https://github.com/audreyr/cookiecutter) and its dependencies:

- cookiecutter
  - jinja2-time
    - arrow
  - whichcraft
  - binaryornot
  - poyo

All of the library tests are running (and passing) except for some in `cookiecutter` itself which require `mercurial`... which can't be sensibly serviced in a "pure" `conda` environment. Also, I assume it's kosher to use pip dependencies inside a test step :)

### CODE REVIEW

File: recipes/arrow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  files:
+    - requirements.txt
+  requires:
+    - nose
+    - coverage
+    - simplejson
+
+about:
+  home: https://github.com/{{ gh_org }}/{{ gh_repo }}
+  license: BSD
```
Comment (msarahan) [inline recipes/arrow/meta.yaml]: 2 clause or 3?

File: recipes/arrow/bld.bat [hunk 2]
```diff
@@ -0,0 +1,2 @@
+python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (msarahan) [inline recipes/arrow/bld.bat]: Can get rid of bld.bat and build.sh and instead in meta.yaml:

```
build:
    script: python setup.py ...
```

File: recipes/poyo/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    - python
+  run:
+    - python
+
+test:
+  requires:
+    - pytest
+
+about:
+  home: https://github.com/{{ gh_org }}/{{ gh_repo }}
```
Comment (msarahan) [inline recipes/poyo/meta.yaml]: 2 clause or 3?

File: recipes/poyo/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  license: BSD
+  summary: >
```
Comment (msarahan) [inline recipes/poyo/meta.yaml]: This block comment seems unnecessary for such a short description

File: recipes/whichcraft/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+test:
+  requires:
+    - flake8
+    - pytest
+    - wheel
+    - pytest-cov
+
+about:
+  home: https://github.com/{{ gh_org }}/{{ gh_repo }}
+  license: BSD
```
Comment (msarahan) [inline recipes/whichcraft/meta.yaml]: Version?

File: recipes/poyo/run_test.sh [hunk 6]
```diff
@@ -0,0 +1,2 @@
+cd "${SRC_DIR}"
```
Comment (msarahan) [inline recipes/poyo/run_test.sh]: Replace with build/script on meta.yaml

File: recipes/poyo/bld.bat [hunk 7]
```diff
@@ -0,0 +1,2 @@
+python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (msarahan) [inline recipes/poyo/bld.bat]: Replace this and build.sh with build/script in meta.yaml

**Comment by bollwyvl** [comment]
Thanks, @msarahan! I think I've incorporated all your comments!

**Comment by bollwyvl** [comment]
Hooray, much 💚 . What's next?

**Comment by msarahan** [comment]
Thanks @bollwyvl

**Comment by jakirkham** [comment]
As the `cookiecutter` builds started around the same time as their dependencies did, many of them failed due to missing dependencies. However, those dependencies are now released so I have restarted those `cookiecutter` builds (and it appears on Mac and Linux that they are released). In the case of AppVeyor, none of the builds had run yet (queue is pretty backed up). However, they were ordered in queue in such a way that they would fail. I reorganized them in the queue so this shouldn't be a problem. Please let us know if you run into any issues.

**Comment by msarahan** [comment]
Thanks @jakirkham for keeping an eye on it.

**Comment by jakirkham** [comment]
Of course.

**Comment by bollwyvl** [comment]
Sorry... didn't anticipate wrapping so many things up at once, ended up
very inception-like. Thanks for jiggering the queue!

**Comment by jakirkham** [comment]
No worries. A little bit of queue jiggering is not really so bad. I think it took longer to write that comment than it did to fix. 😆

At least with Travis CI and CircleCI, it will give you privileges to restart. So, if you run into issues with those, it is easy to fix. AppVeyor doesn't work like that unfortunately. If it is AppVeyor related, feel free to either ping us to fix it. If we can't fix it fast enough for some reason, you can always PRs with no net change to affect a restart.

Also, not sure if we had this conversation (just trying to let everyone know this), make sure when you want to make a change: fork the repo, make the change in your fork, and then PR it back to the main repo. Changes made to the main repo are currently automatically deployed. We are planning to change this a bit in the future. ( https://github.com/conda-forge/conda-forge.github.io/issues/87 )

**Pull request merged**